Sources/PaymentButtons/PayPalButton.swift

```diff
@@ -113,69 +113,65 @@ public final class PayPalButton: PaymentButton {
-        ) {
-            button = PayPalButton(
-                fundingSource: .payPal,
-                color: color.color,
-                edges: edges,
-                size: size,
-                insets: insets,
-                label: label?.label
-            )
-            self.action = action
-        }
-       
-        // MARK: - UIViewRepresentable methods
-        // TODO: Make unit test for UIVRepresentable methods: https://engineering.paypalcorp.com/jira/browse/DTNOR-623
+/// PayPalButton for SwiftUI
+public struct PayPalButtonView: UIViewRepresentable {
 
-        public func makeCoordinator() -> Coordinator {
-            Coordinator(action: action)
-        }
+    private var action: () -> Void = { }
+
+    private var button: PayPalButton
+
+    /// Initialize a PayPalButton
+    /// - Parameters:
+    ///   - insets: Edge insets of the button, defining the spacing of the button's edges relative to its content.
+    ///   - color: Color of the button. Default to gold if not provided.
+    ///   - edges: Edges of the button. Default to softEdges if not provided.
+    ///   - size: Size of the button. Default to standard if not provided.
+    ///   - label: Label displayed next to the button's logo. Default to no label.
+    public init(
+        insets: NSDirectionalEdgeInsets? = nil,
+        color: PayPalButton.Color = .gold,
+        edges: PaymentButtonEdges = .rounded,
+        size: PaymentButtonSize = .standard,
+        label: PayPalButton.Label? = nil,
+        _ action: @escaping () -> Void = { }
+    ) {
+        button = PayPalButton(
+            fundingSource: .payPal,
+            color: color.color,
+            edges: edges,
+            size: size,
+            insets: insets,
+            label: label?.label
+        )
+        self.action = action
+    }
 
-        public func makeUIView(context: Context) -> UIView {
-            let view = UIView()
+    // MARK: - UIViewRepresentable methods
+    // TODO: Make unit test for UIVRepresentable methods: https://engineering.paypalcorp.com/jira/browse/DTNOR-623
 
-            view.addSubview(button)
-            button.addTarget(context.coordinator, action: #selector(Coordinator.onAction(_:)), for: .touchUpInside)
+    public func makeCoordinator() -> Coordinator {
+        Coordinator(action: action)
+    }
 
-            return view
-        }
+    public func makeUIView(context: Context) -> PaymentButton {
 
-        public func updateUIView(_ uiView: UIView, context: Context) {
-            context.coordinator.action = action
-        }
+        let button = button
+        button.addTarget(context.coordinator, action: #selector(Coordinator.onAction(_:)), for: .touchUpInside)
+        return button
+    }
+
+    public func updateUIView(_ uiView: PaymentButton, context: Context) {
+        context.coordinator.action = action
     }
 }
 
 // MARK: PayPalButton Preview
 
-struct PayPalButtonView: View {
+struct PayPalButtonUIView: View {
```

Comment:
Do we need this view? I see that we are referencing PayPalCreditButtonView in demo app. Same question for the Credit and Pay Later buttons.

Reply:
Follow-up: Correct me if I'm wrong, but I believe we saw instances where we needed to call `.fixedSize(...)` in the demo app. If that's the case, should we use this wrapper view in the demo app instead of the `PayPalButtonView` so as to eliminate the merchant's need to call `.fixedSize(...)` themselves?

Reply:
The view was here before and is used so we can preview the button element on its own, separate from when it's used with other elements in `SwiftUIPaymentButtonDemo`. In terms of if we need it, technically we don't since we have the demo app to see the button.

afaik, we can't call `PayPalButtonUIView` in `SwiftUIPaymentButtonDemo` because the latter is expecting a UIViewRepresentable and now a View. I did try to do so just now and I'm not able to add `PayPalButtonUIView`. 

I think Merchant devs would know to used `.fixedSize()` since it's a common issue in SwiftUI, but in case they don't, I can add that to the dev doc sample code. What do you think?

Reply:
Oh I see, this is the pre-preview struct. Can we rename these preview structs to indicate they they are meant to be for the preview? Also, can we move the struct inside the preview struct as well?

Example:

```
struct PayPalButtonView_Preview: PreviewProvider {

    static var previews: some View {
        PayPalButtonView()
    }
    
    private struct PayPalButtonWrapper: View {
        var body: some View {
            PayPalButton.Representable()
        }
    }
}
```

Reply:
Oops I just saw this. I'll make the change!